hudi-client/src/main/java/org/apache/hudi/table/action/commit/BaseCommitActionExecutor.java

```diff
@@ -230,6 +231,9 @@ protected void commit(Option<Map<String, String>> extraMetadata, HoodieWriteMeta
     metadata.addMetadata(HoodieCommitMetadata.SCHEMA_KEY, getSchemaToStoreInCommit());
     metadata.setOperationType(operationType);
 
+    // Update Metadata Table
+    HoodieMetadata.update(config, metadata, instantTime);
```

Comment:
Note to self: need to understand how this call is different from the call made in writeClient code

Reply:
I observed that certain commit paths do not go via the HoodieWriteCLient  (HoodieWriteClient.completeCompaction(), AbstractHoodieClient.commitStats()) as they have been moved into the various executors. Hence, I could not find a single location which would be sufficient to issue the  HoodieMetadata.update.

This is also the reason I could not use a member variable in HoodieWriteClient as you had suggested elsewhere.